src/core/meshcop/tcat_agent.cpp

```diff
@@ -523,6 +533,52 @@ Error TcatAgent::HandleSetActiveOperationalDataset(const Message &aIncomingMessa
 
     Get<ActiveDatasetManager>().SaveLocal(dataset);
 
+    SuccessOrExit(error = otBleSecureGetPeerCertificateRaw(&GetInstance(), buf, &bufLen));
+    Get<Settings>().SaveTcatCommissionerCertificate(buf, (uint16_t)bufLen);
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleGetActiveOperationalDataset(Message &aOutgoingMessage, bool &aResponse)
+{
+    Error         error = kErrorNone;
+    Dataset       dataset;
+    Dataset::Tlvs datasetTlvs;
+
+    if (!CheckCommandClassAuthorizationFlags(mCommissionerAuthorizationField.mApplicationFlags,
+                                             mDeviceAuthorizationField.mApplicationFlags, &dataset))
```

Comment:
The TLV type Get Active Operational Dataset resides in the command class extraction not command class application.
I should say:
if (!CheckCommandClassAuthorizationFlags(mCommissionerAuthorizationField.mExtractionFlags,
                                             mDeviceAuthorizationField.mExtractionFlags, &dataset))

Reply:
fixed.